src/Config/RuntimeConfigLoader.cs

```diff
@@ -30,23 +31,36 @@ public abstract class RuntimeConfigLoader
     // state in place of using out params.
     public RuntimeConfig? RuntimeConfig;
 
-    // Signals a hot reload event for OpenApiDocumentor due to config change.
-    protected virtual void DocumentorOnConfigChanged(HotReloadEventArgs args)
+    public RuntimeConfigLoader(HotReloadEventHandler<HotReloadEventArgs>? handler = null, string? connectionString = null)
     {
-        _handler?.DocumentorOnConfigChangedEvent(this, args);
+        _handler = handler;
+        _connectionString = connectionString;
     }
 
-    // Sends all of the notifications when a hot reload occurs.
-    public void SendEventNotification(string message = "")
+    protected virtual void OnConfigChangedEvent(HotReloadEventArgs args)
     {
-        HotReloadEventArgs args = new(message);
-        DocumentorOnConfigChanged(args);
+        _handler?.OnConfigChangedEvent(this, args);
     }
 
-    public RuntimeConfigLoader(HotReloadEventHandler<HotReloadEventArgs>? handler = null, string? connectionString = null)
+    /// <summary>
+    /// Sends the notification to the event handler to trigger the hot-reload events
+    /// that have subscribed for configuration changes. The order here matters since
+    /// there are dependencies that we must refresh one after another. If adding to this
+    /// method make sure that you add the event trigger after any required dependencies have
+    /// been refreshed by previously called event triggers.
+    /// </summary>
+    /// <param name="message"></param>
+    public void SendEventNotification(string message = "")
```

Comment:
Can this be protected because it is only called from `FileSystemRuntimeConfigLoader::HotReloadConfig(...)` and I don't think we intend for other components to be able to call `SendNotification(...)`